File: server/models/message.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const MessageSchema = Schema({
    sender: {type: mongoose.Schema.Types.ObjectId, ref: "User"},
    content: {type: String, trim: true},
    chat: {type: mongoose.Schema.Types.ObjectId, ref: "Chat"},
    isSeen: {type: Boolean, default: false},
    image: {type: String, default: ''}
},
{
    timestamps:true
});

module.exports = mongoose.model("Message", MessageSchema);

File: ServerNET/Domain/Message.cs
using System;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Domain;

public class Message
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    [BsonElement("_id")]
    public string? Id { get; set; }

    [BsonRepresentation(BsonType.ObjectId)]
    [BsonElement("sender")]
    public string Sender { get; set; } = null!;

    [BsonElement("content")]
    public string Content { get; set; } = null!;

    [BsonRepresentation(BsonType.ObjectId)]
    [BsonElement("chat")]
    public string Chat { get; set; } = null!;

    [BsonElement("isSeen")]
    public bool IsSeen { get; set; }

    [BsonElement("image")]
    public string? Image { get; set; }

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; }

    [BsonElement("updatedAt")]
    public DateTime? UpdatedAt { get; set; }
}
